crates/contract/tests/sandbox/common.rs

```diff
@@ -988,30 +989,30 @@ pub async fn call_contract_key_generation<const N: usize>(
         }
     };
 
-    for (domain_counter, domain) in domains_to_add.iter().enumerate() {
```

Comment:
this refactor is mostly an optimization, a bit orthogonal to the PR, but seemed necessary at first